src/react/databrowser/objects/ObjectListTable.jsx

```diff
@@ -94,10 +94,13 @@ export default function ObjectListTable({
         ),
       );
     }
-    // We exclude `versionId` from the dependency array,
-    // because versionId will be patched to the query params in the property tab.
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [dispatch, isListVersion, objectsLength, objectKey]);
+  }, [
+    dispatch,
+    isListVersion,
+    objectsLength,
+    objectKey,
+    isListVersion ? versionId : null,
```

Comment:
`isListVersion, versionId`

Reply:
OK. I think it would be easier to understand in this way.
Actually, we need to add a conditional dependency `versionId` in the dependency array.

The reason is that we are not listing versions, `versionId` should not appear in the dependency array. Otherwise when we patch the `versionId` to the URL query params it will trigger another **toggleObject** action with the versionId, where we don't have this object in the list..

```js
  const versionIdOrUndefined =
    isListVersion && versionId ? versionId : undefined;
  useEffect(() => {
    if (objectsLength > 0) {
      dispatch(
        toggleObject(
          isFolder ? `${objectKey}/` : objectKey,
          versionIdOrUndefined,
        ),
      );
    }
  }, [
    dispatch,
    isListVersion,
    objectsLength,
    objectKey,
    isFolder,
    versionIdOrUndefined,
  ]);
 ```